hawkular-alerts-engine/src/main/resources/org/hawkular/alerts/engine/rules/ConditionMatch.drl

```diff
@@ -621,6 +621,156 @@ rule SetFiringModeOnSatisfiedDampening
         $t.setMode(Mode.FIRING);
         autoResolvedTriggers.put( $t, $d.getSatisfyingEvals() );
 
-        retract( $d );
         retract( $t );
 end
+
+
+////// TRIGGER RETRACTION CLEANUP
+
+// If a trigger is retracted these rules will ensure all associated facts are removed. This only assures
+// a clean working memory if the trigger is retracted while rules are firing. If a Trigger fact is removed via
+// the API while the rule-base is not firing, then these rules don't fire and manual cleanup may be necessary.
+
+rule retractDampening
+    when
+        $d : Dampening( $tenantId : tenantId, $tid : triggerId )
+        not  Trigger( tenantId == $tenantId, id == $tid )
+    then
+        retract( $d )
+
+        if (log != null && log.isDebugEnabled()) {
+            log.debugf("Retract Dampening with no Trigger. %s", $d);
+        }
+end
+
+rule retractAvailabilityCondition

```

Comment:
Yes of course!  I am an idiot, for whatever reason I was thinking Drools wouldn't deal with the abstract class. Fixing this...